### PR TITLE
Heedls 422 delegate approvals

### PR DESCRIPTION
**Pull request by JBsoftwire**
First phase review for HEEDLS-422, with foundational page elements but not yet loading or displaying specific delegate data.

### CODE REVIEW

**Review by DanBloxham-sw — APPROVED** [review]
Looks good to me so far.

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public DelegateApprovalsViewModel(IEnumerable<ApprovableDelegate> delegates)
+        {
+            Delegates = delegates;
+        }
+
+        public IEnumerable<ApprovableDelegate> Delegates { get; set; }
+    }
+
+    public class ApprovableDelegate
```
Comment (stellake) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs:17]: Could this be UnapprovedDelegate?

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public int Id { get; set; }
+        public string FirstName { get; set; }
```
Comment (stellake) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs:20]: I wonder if we should have FullName field here and we set it in the constructor later (to move the logic from HTML to C# code)

File: DigitalLearningSolutions.Web/Views/MyAccount/MyDetailsCustomField.cshtml [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+@model CustomFieldViewModel
+
+<div class="nhsuk-summary-list__row details-list-with-button__row">
+  <dt class="nhsuk-summary-list__key">
+    @Model.CustomPrompt@(Model.Mandatory ? ":" : " (optional):")
+  </dt>
+  <dd class="nhsuk-summary-list__value">
+    @Model.Answer
+  </dd>
+</div>
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Views/MyAccount/MyDetailsCustomField.cshtml:11]: Looks like this is changed to unix line endings - have you got the correct configuration for line endings?

File: DigitalLearningSolutions.Web/Views/TrackingSystem/DelegateApprovals/Index.cshtml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  ViewData["Title"] = "Approve delegate registrations";
+}
+
+<div class="nhsuk-grid-row">
+  <div class="nhsuk-grid-column-full">
+    <h1 class="nhsuk-heading-xl">Approve delegate registrations</h1>
+    @if (Model.Delegates.Any()) {
+      @foreach(var delegateApproval in Model.Delegates)
+      {
+        <partial name="_delegateExpandable" model="delegateApproval" />
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/DelegateApprovals/Index.cshtml:14]: Could this partial be called _approveDelegateExpandable (we will have other delegate expandables unrelated to the approval functionality)

File: DigitalLearningSolutions.Web/Views/TrackingSystem/DelegateApprovals/Index.cshtml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          <p>No accounts need approval.</p>
+        </div>
+      </div>
+      <div class="nhsuk-grid-row">
+        <div class="nhsuk-grid-column-full">
+          <a class="nhsuk-button" href="" role="button">
+            Manage approved delegates
+          </a>
+        </div>
+      </div>
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/DelegateApprovals/Index.cshtml:35]: Could this be an NHS action link (we should have a VC for it) https://service-manual.nhs.uk/design-system/components/action-link

Can we also link to [old url] + /tracking/delegates?

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/DelegateApprovalsController.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/DelegateApprovalsController.cs:1]: Could we add this page to automated accessibility tests?

Comment (stellake) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/DelegateApprovalsController.cs:1]: Could we put this + views into a folder called "Delegates" as we'll have several delegate related controllers under the Tracking system?

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateApprovals/Index.cshtml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        <div class="nhsuk-grid-column-full">
+          <div class="nhsuk-action-link">
+            <a class="nhsuk-action-link__link" href="@Configuration["CurrentSystemBaseUrl"]/tracking/delegates">
+              <svg class="nhsuk-icon nhsuk-icon__arrow-right-circle" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true">
+                <path d="M0 0h24v24H0z" fill="none"></path>
+                <path d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z"></path>
+              </svg>
+              <span class="nhsuk-action-link__text">Manage approved delegates</span>
+            </a>
+          </div>
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateApprovals/Index.cshtml:41]: Ah just realised we can't use a VC due to the href.. I guess we can change it when we create the delegates page :)

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem
```
Comment (stellake) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/DelegateApprovalsViewModel.cs:1]: Should this be in the Delegates folder too? :)

**Review by stellake — APPROVED** [review]
Looks great! 👍

**Pull request merged**